### DISCUSSION TITLE
`slam_toolbox` not generating a map

### DISCUSSION THREAD

**Discussion by sebastianfer** [iRobotEducation/create3_docs#123] [Questions]
I am trying to generate a map, but rviz2 is not showing it. SLAM seems to be working fine; I tested it, and the red dots are showing, but when I move the Create, a map doesn't show up. I am using iRobot's files, and I edited the data for the camera I am using (RPLidar A2M8). I also can see map under the topic list. 

I am also attaching pictures of the TF tree if it helps. 

![Screen Shot 2022-07-30 at 6 19 50 PM](https://user-images.githubusercontent.com/108418842/182002439-09521f67-2848-4cb9-9cc8-bd31bcbabbde.png)
![Screen Shot 2022-07-30 at 6 19 34 PM](https://user-images.githubusercontent.com/108418842/182002440-d47ea55a-bc93-4e34-83bb-756bcf8c9410.png)

Any idea of why this is happening?

**Comment by justinIRBT** [comment]
Can you try seeing if the map topic is showing on your SBC (raspberry pi, if that is what the SBC is).  On your SBC you can try
ros2 topic hz /map
ros2 topic bw /map
ros2 topic echo /map —no-arr
You can also try /map_updates topic.

can you also post the output from the slam_toolbox terminal

**Reply by sebastianfer** [reply]
The following topics are being published in my SBC:
/map 
/map_metadata
/map_updates 

Out of all of the commands you listed, the only one that returned something was ros2 topic bw /map; I got: Subscribed to [/map]

This is the output of my Slam_toolbox terminal: 
[INFO] [launch]: All log files can be found below /home/pi/.ros/log/2022-07-31-11-09-10-475848-ubuntu-7558
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [async_slam_toolbox_node-1]: process started with pid [7561]
[async_slam_toolbox_node-1] [INFO] [1659280151.247748760] [slam_toolbox]: Node using stack size 40000000
[async_slam_toolbox_node-1] [INFO] [1659280151.481648915] [slam_toolbox]: Using solver plugin solver_plugins::CeresSolver
[async_slam_toolbox_node-1] [INFO] [1659280151.484524190] [slam_toolbox]: CeresSolver: Using SCHUR_JACOBI preconditioner.
[async_slam_toolbox_node-1] [INFO] [1659280151.527248266] [slam_toolbox]: Message Filter dropping message: frame 'laser_frame' at time 1659280151.419 for reason 'discarding message because the queue is full'
[async_slam_toolbox_node-1] [INFO] [1659280151.579066812] [slam_toolbox]: Message Filter dropping message: frame 'laser_frame' at time 1659280151.499 for reason 'discarding message because the queue is full'
[async_slam_toolbox_node-1] [INFO] [1659280151.659026430] [slam_toolbox]: Message Filter dropping message: frame 'laser_frame' at time 1659280151.579 for reason 'discarding message because the queue is full'
[async_slam_toolbox_node-1] [WARN] [1659280151.704220932] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280151.803561203] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280151.865186641] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280151.958698825] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280152.008586386] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280152.104712662] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280152.204183563] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280152.254681285] [slam_toolbox]: Failed to compute odom pose
[async_slam_toolbox_node-1] [WARN] [1659280152.362174182] [slam_toolbox]: Failed to compute odom pose


I also saved the frames this last time, and this is what I got: 
[frames.pdf](https://github.com/iRobotEducation/create3_docs/files/9229142/frames.pdf)

**Reply by justinIRBT** [reply]
Does it keep saying “Failed to compute odom pose” or does it stop after a while?  When you are republishing the transforms from your /create/ namespace to just the frame name, are you passing the timestamp from one to the other?  Your frame tree looks fine, but I wonder if the timestamps are matching the /scan timestamp.

**Reply by sebastianfer** [reply]
I just tried again, and it keeps saying it. I also echoed the topic /scan and /odom, and I got these results (Green is odom, and white is scan):
![Screen Shot 2022-08-01 at 10 30 47 AM](https://user-images.githubusercontent.com/108418842/182172935-e747d239-768e-488f-9be0-0962e4e2cdf2.png)

![Screen Shot 2022-08-01 at 10 30 39 AM](https://user-images.githubusercontent.com/108418842/182172798-6b0131b3-76ba-4263-b199-2baf3738238f.png)

Also, I ran ros2 run tf2_ros tf2_echo odom base_link, and this is what I got: At time 1659364192.466555235
- Translation: [0.051, -0.071, 0.012]
- Rotation: in Quaternion [-0.001, -0.012, 0.928, 0.373]
At time 1659364193.458927100
- Translation: [0.051, -0.071, 0.012]
- Rotation: in Quaternion [-0.001, -0.012, 0.928, 0.373]
At time 1659364194.467523988
- Translation: [0.051, -0.071, 0.012]
- Rotation: in Quaternion [-0.001, -0.012, 0.928, 0.373]
At time 1659364195.458417229
- Translation: [0.051, -0.071, 0.012]
- Rotation: in Quaternion [-0.001, -0.012, 0.928, 0.373]


After running ros2 run tf2_ros tf2_monitor:

RESULTS: for all Frames

Frames:
Frame: base_footprint, published by <no authority available>, Average Delay: 0.0423885, Max Delay: 0.328891
Frame: base_link, published by <no authority available>, Average Delay: 0.0423868, Max Delay: 0.328887
Frame: bump_front_center, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_1, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_2, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_power, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_omni, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: laser_frame, published by <no authority available>, Average Delay: 24.6188, Max Delay: 24.6188
Frame: left_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: mouse, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: odom, published by <no authority available>, Average Delay: -0.0855761, Max Delay: 0.0891907
Frame: right_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9

All Broadcasters:
Node: <no authority available> 70.7492 Hz, Average Delay: 116.01 Max Delay: 85162.9



RESULTS: for all Frames

Frames:
Frame: base_footprint, published by <no authority available>, Average Delay: 0.203647, Max Delay: 2.62037
Frame: base_link, published by <no authority available>, Average Delay: 0.203646, Max Delay: 2.62037
Frame: bump_front_center, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_1, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_2, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_power, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_omni, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: laser_frame, published by <no authority available>, Average Delay: 24.6188, Max Delay: 24.6188
Frame: left_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: mouse, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: odom, published by <no authority available>, Average Delay: 0.0773258, Max Delay: 2.5054
Frame: right_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9

All Broadcasters:
Node: <no authority available> 70.1111 Hz, Average Delay: 0.181182 Max Delay: 2.62037



RESULTS: for all Frames

Frames:
Frame: base_footprint, published by <no authority available>, Average Delay: 0.148206, Max Delay: 2.62037
Frame: base_link, published by <no authority available>, Average Delay: 0.148204, Max Delay: 2.62037
Frame: bump_front_center, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: bump_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_1, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_2, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: button_power, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: cliff_side_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_center_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_front_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_right, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_intensity_side_left, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: ir_omni, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: laser_frame, published by <no authority available>, Average Delay: 24.6188, Max Delay: 24.6188
Frame: left_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: mouse, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9
Frame: odom, published by <no authority available>, Average Delay: 0.0737716, Max Delay: 2.5054
Frame: right_wheel, published by <no authority available>, Average Delay: 85162.9, Max Delay: 85162.9

All Broadcasters:
Node: <no authority available> 72.089 Hz, Average Delay: -0.0323204 Max Delay: 2.5054

**Reply by justinIRBT** [reply]
There is a 4 second difference between the scan timestamp and the Odom timestamp, we’re those screenshots taken at approximately the same time?  Remind me what your setup is, is create on wifi and raspberry pi on wifi or are they connected over usb ip?

**Comment by jerrynsr** [comment]
When I looked at RVIZ display that was posted it appears that the orientation of the LIDAR might be wrong.  Make sure the motor and pulley are facing the front of the robot.

**Reply by shamlian** [reply]
I agree with this thought.
